p2p/simulations/adapters/inproc_test.go

```diff
@@ -140,46 +105,35 @@ func TestNetPipe(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	done := make(chan struct{})
+	msgs := 50
+	size := 1024
+	var wg sync.WaitGroup
+	defer wg.Wait()
 
+	// netPipe is blocking, so writes are emitted asynchronously
+	wg.Add(1)
 	go func() {
-		msgs := 50
-		size := 1024
-		// netPipe is blocking, so writes are emitted asynchronously
-		go func() {
-			for i := 0; i < msgs; i++ {
-				msg := make([]byte, size)
-				_ = binary.PutUvarint(msg, uint64(i))
-
-				_, err := c1.Write(msg)
-				if err != nil {
-					t.Fatal(err)
-				}
-			}
-		}()
+		defer wg.Done()
 
 		for i := 0; i < msgs; i++ {
 			msg := make([]byte, size)
-			_ = binary.PutUvarint(msg, uint64(i))
-
-			out := make([]byte, size)
-			_, err := c2.Read(out)
-			if err != nil {
-				t.Fatal(err)
-			}
-
-			if !bytes.Equal(msg, out) {
-				t.Fatalf("expected %#v, got %#v", msg, out)
+			binary.PutUvarint(msg, uint64(i))
+			if _, err := c1.Write(msg); err != nil {
+				t.Error(err)
```

Comment:
Are you sure about this one? I thought that using `t` in a goroutine was creating a problem.

Reply:
calling `t.Errorf` in a separate goroutine is OK